cvmfs/dns.cc

```diff
@@ -1145,44 +1145,61 @@ void HostfileResolver::ParseHostFile() {
   while (GetLineFile(fhosts_, &line)) {
     char address[kIpMaxLength + 1];
     char hostname[kHostnameMaxLength + 1];
-    int bytes_read = 0;
+    int bytes_read;
     size_t str_offset = 0;
 
     // strip comments
     size_t hash_pos = line.find_first_of('#');
     if (hash_pos != string::npos) line = line.substr(0, hash_pos);
 
     // First token is an IP address
-    int ip_start_pos, ip_end_pos;
-    sscanf(line.c_str(), " %n%*s%n", &ip_start_pos, &ip_end_pos);
+    int ip_start_pos = -1, ip_end_pos = -1, scan_result;
+    scan_result = sscanf(line.c_str(), " %n%*s%n", &ip_start_pos, &ip_end_pos);
+    if (scan_result == EOF)
+      continue;
+    assert(ip_start_pos != -1);
+    assert(ip_end_pos != -1);
     if (ip_start_pos == ip_end_pos)
       continue;
     if (ip_end_pos - ip_start_pos > kIpMaxLength) {
       LogCvmfs(kLogDns, kLogSyslogWarn,
-               "Skipping line in hosts file due to invalid IP address format");
+        string("Skipping line in hosts file due to invalid IP address format: "
```

Comment:
It would be more natural to use the form

    "Skipping line in hosts file due to invalid IP address format: %s", line.c_str()

Reply:
Ah, sorry, I totally forgot about this possibility.